java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -310,6 +310,37 @@ public BaseDeviceMemoryBuffer getDeviceBufferFor(BufferType type) {
     return srcBuffer;
   }
 
+  /**
+   * Create a ColumnVector from the off heap Apache Arrow buffers passed in.
+   * @param type - type of the column
+   * @param colName - Name of the column
+   * @param numRows - Number of rows in the arrow column
+   * @param nullCount - Null count
+   * @param data - address of the Arrow data buffer
+   * @param dataLength - size of the Arrow data buffer
+   * @param validity - address of the Arrow validity buffer
+   * @param validityLength - size of the Arrow validity buffer
+   * @param offsets - address of the Arrow offsets buffer
+   * @param offsetsLength - size of the Arrow offsets buffer
+   * @return - new ColumnVector
+   */
+  public static ColumnVector fromArrow(
+      DType type,
+      String colName,
+      long numRows,
+      long nullCount,
+      long data,
```

Comment:
This API would be more Java-like if it took `ByteBuffer` instances (which can be `DirectByteBuffer` for off-heap).  Not sure if that makes it easier to wield for the intended use-case.

Reply:
thanks for pointing me to the jni bytebuffer api, I wasn't aware of a way to get the address and size